### PR TITLE
Fix main detection for debug PE bins ##bin

### LINKED ISSUE #7721: r2 fails to detect main
the bin is compiled with VS C++ 2016 on win10 and uses the MSVCP runtime dll.
attached the really simple bin.
consoleapplication1

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Closing issues**

Fixes #7721

### CODE REVIEW

**Comment by XVilka** [comment]
Could you please add a test for this?

File: libr/bin/format/pe/pe.c [hunk 1]
```diff
@@ -125,33 +142,110 @@ struct r_bin_pe_addr_t *PE_(check_msvcseh) (struct PE_(r_bin_pe_obj_t) *bin) {
...
 			}
+		}
+	}
 
+	// MSVC 32bit debug 
+	// push ebp
+	// mov ebp, esp
+	// call xxxxxxxx
+	if (b[3] == 0xe8) {
+		const st32 call_dst = r_read_ble32 (b + 4, bin->big_endian);
```
Comment (XVilka) [inline libr/bin/format/pe/pe.c:171]: We probably should make a macro or a function for this pattern.

File: libr/bin/format/pe/pe.c [hunk 2]
```diff
@@ -125,33 +142,110 @@ struct r_bin_pe_addr_t *PE_(check_msvcseh) (struct PE_(r_bin_pe_obj_t) *bin) {
...
+		entry->paddr += (5 + 3 + call_dst);
+		entry->vaddr += (5 + 3 + call_dst);
+		r_buf_read_at (bin->b, entry->paddr, b, sizeof (b));
+		if (b[8] == 0xe8) {
```
Comment (ret2libc) [inline libr/bin/format/pe/pe.c:175]: could you add a comment with the pattern matched by this series of `if`?

File: libr/bin/format/pe/pe.c [hunk 3]
```diff
@@ -125,33 +142,110 @@ struct r_bin_pe_addr_t *PE_(check_msvcseh) (struct PE_(r_bin_pe_obj_t) *bin) {
...
-		if (r_buf_read_at (bin->b, entry->paddr, b, sizeof (b)) > 0) {
+	if (b[4] == 0xe8) {
+		bool found_caller = false;
+		if (b[13] == 0xe9) {
+			const st32 jmp_dst = r_read_ble32 (b + 14, bin->big_endian);
+			entry->paddr += (5 + 13 + jmp_dst);
+			entry->vaddr += (5 + 13 + jmp_dst);
+			found_caller = r_buf_read_at (bin->b, entry->paddr, b, sizeof (b)) > 0;
+		} else {
+			const st32 call_dst = r_read_ble32 (b + 5, bin->big_endian);
```
Comment (ret2libc) [inline libr/bin/format/pe/pe.c:209]: The comment at https://github.com/radareorg/radare2/pull/17376/files#diff-15f63750bbf1cea750d78c8cfc33a916R196 seems to refer to the `if` case. What is the `else` matching exactly? Could you add a comment for this as well?

**Pull request merged**